code/game/machinery/vending/cm_vending.dm

```diff
@@ -1366,7 +1367,7 @@ GLOBAL_LIST_INIT(cm_vending_gear_corresponding_types_list, list(
 	if(LAZYLEN(itemspec)) //making sure it's not empty
 		if(vend_delay)
 			overlays.Cut()
-			icon_state = "[initial(icon_state)]_vend"
+			flick("[initial(icon_state)]_vend", src)
```

Comment:
`/obj/structure/machinery/cm_vending/proc/vend_fail()` must also then have a valid icon state or some other amended implementation. See https://github.com/cmss13-devs/cmss13/blob/master/code/game/machinery/vending/cm_vending.dm#L830-L834
![vend fail](https://github.com/user-attachments/assets/90d8f9a6-b79a-47a6-88d2-5a0f43facba4)

Reply:
i forgot to remove global vend delay that i used for testing, i'll all animations to all vendors, just a bit later and in a separate pr